src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -14836,6 +14975,13 @@ public Base(int X, int Y)
             Assert.Same("<global namespace>", model.GetEnclosingSymbol(x.SpanStart).ToTestDisplayString());
             Assert.Empty(model.LookupSymbols(x.SpanStart, name: "X"));
             Assert.DoesNotContain("X", model.LookupNames(x.SpanStart));
+
+            var recordDeclarations = tree.GetRoot().DescendantNodes().OfType<RecordDeclarationSyntax>().Skip(1).ToArray();
+
+            Assert.Equal("C", recordDeclarations[0].Identifier.ValueText);
+            Assert.Null(model.GetOperation(recordDeclarations[0]));
```

Comment:
>model.GetOperation(recordDeclarations[0]) [](start = 24, length = 41)

Why does `GetOperation()` return `null`?

Reply:
> Why does GetOperation() return null?

There is no user code for this primary constructor, no code blocks to fire analyzers on, we never create a member semantic model for a primary constructor like that, etc. Technically, we could produce an "empty" IConstructorBodyOperation node, but then we would have to do something with analyzers to make sure an action for this node would fire. Unfortunately, analyzer driver is able to do that only when there is a "real" node for an initializer or a body (see ```GetOperationsToAnalyze``` in AnalyzerDriver.cs), we have neither of them here. I decided to not deal with this complexity right now, when we don't really have a real scenario to enable.  

---
In reply to: [461876729](https://github.com/dotnet/roslyn/pull/46308#discussion_r461876729) [](ancestors = 461876729)